runtime/include/qio/qio_popen.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2004-2015 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef QIOPLUGIN_POPEN_H_
+#define QIOPLUGIN_POPEN_H_
+
+#include "sys_basic.h"
+#include "qio.h"
+#include <stdarg.h>
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
It allows the QIO headers to be included from a C++ program. C doesn't have an extern "C".